### PR TITLE
state: add trainingStore,modelStore. update: userStore

### PR DESCRIPTION
**Pull request by techyNonso**
### Description

- [x] Relates #130 

### Checklist

- [x] I have followed the repository's contribution guidelines.
- [x] I have added necessary documentation or updated existing documentation.
- [x] My code follows the established code style and formatting of the project.
- [x] I have tested my changes and they function as expected.
- [x] I have added suitable unit tests (if applicable).

To be reviewed by: @Nasfame @techyNonso

### CODE REVIEW

File: src/state/modelStore.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { create } from 'zustand'
+import { devtools, persist } from 'zustand/middleware'
+import { upsert_user } from '@app/explore/upsert_user'
+import { Model, TrainingRequest } from '@prisma/client'
+
+type IModel = Omit<Omit<Model, 'createdAt'>, 'updatedAt'>
+type ITrainRequest = Omit<Omit<TrainingRequest, 'createdAt'>, 'updatedAt'>
+
+type Store = {
+  model: IModel | null
```
Comment (Nasfame) [inline src/state/modelStore.ts:10]:
```
type Store = {
  user: IUser | null
  init: (user: IUser) => void
  setUser: (user: Partial<IUser>) => void
}

```

refer the store above its maintaining user state and keeping it in sync with "user" collection in DB

Similarily for every collection in mongoDB, there should be a store. 

It should be like: the stores are our primariy DB and the changes are replicated to mongoDB

Comment (Nasfame) [reply src/state/modelStore.ts:10]: so there needs to be separate **trainRequest** Store

Comment (Nasfame) [reply src/state/modelStore.ts:10]: for ModelStore: on init: it should get all models of userId from DB. 

Don't use :   models: IModel[] | null

It becomes harder to access. Use an object : {modelID: model}

Good job on creating:   model: IModel | null to maintain the state of DB-unsynced model. It does raise concerns if the user opens different tabs and tries to create models in new tabs.

Comment (Nasfame) [reply src/state/modelStore.ts:10]: but that's alright, we can restrict user-usage to just one-tab I guess?

**Comment by techyNonso** [comment]
Let's get on a call later cos I don't understand what you are saying.

File: src/state/modelStore.ts (hunk 1 above)
Comment (techyNonso) [reply src/state/modelStore.ts:10]: > for ModelStore: on init: it should get all models of userId from DB.
> 
> Don't use : models: IModel[] | null
> 
> It becomes harder to access. Use an object : {modelID: model}
> 
> Good job on creating: model: IModel | null to maintain the state of DB-unsynced model. It does raise concerns if the user opens different tabs and tries to create models in new tabs.

I do not understand this object section

Comment (Nasfame) [reply src/state/modelStore.ts:10]: Ok lets get on the call

**Review by Nasfame — APPROVED** [review]
LGTM

**Pull request merged**